google-cloud-logging/lib/google/cloud/logging/logger.rb

```diff
@@ -121,7 +171,8 @@ def progname= name
         # @deprecated Use request_info
         #
         def trace_ids
-          @request_info.inject({}) { |r, (k, v)| r[k] = v.trace_id }
+          request_info_dup = synchronize { @request_info_map.dup }
+          request_info_dup.inject({}) { |r, (k, v)| r[k] = v.trace_id }
```

Comment:
Instead of duplicating, why not just wrap the inject invocation in a synchronize block and be done with it?

Reply:
If we don't synchronize access to @writer, should we then also make AsyncWriter thread-safe?

Reply:
I had it that was originally, but I moved the call to `inject` outside of the synchronize` block because `inject` isn't the fastest method. If a hash has 1,000 pairs, it didn't seem like it was smart to lock up the logger for that method call.

Reply:
I'd be surprised if the difference is great enough to matter. And even if the difference is not insignificant, we should only be blocking other operations that manipulate `@request_info_map`, which shouldn't be that many (which is why we should keep synchronization to the minimum necessary.)